### ISSUE TITLE
[Accessibility] Narrator cannot announce the indeterminate state of CheckedListBox item correctly

### ISSUE THREAD

**Issue by v-elnovikova** [dotnet/winforms#9243]
### .NET version

.NET SDK 8.0.100-preview.4.23260.5

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

Yes, it is a regression issue, cannot repro in .Net 5, and repro in .Net 6, 7 versions

### Issue description

CheckedListBox item's state can be set to **Indeterminate** programmatically:

```
checkedListBox1.SetItemCheckState(2, CheckState.Indeterminate);
```

Narrator announces **Indeterminate** state as **Checked**.

#### .NET 8.0:
![CheckedListBox - indeterminate state annouced as checked in 8 0](https://github.com/dotnet/winforms/assets/113603457/17beb2a1-bd9c-43d7-bb54-9c6e2c1d63f5)
![CheckedListBox - indeterminate state as On in 8 0](https://github.com/dotnet/winforms/assets/113603457/4cb60f37-83a3-4aec-ad15-8352f72163e0)

#### .NET 5.0:
![CheckedListBox - indeterminate state annouced correctly in 5 0](https://github.com/dotnet/winforms/assets/113603457/3d2b8fda-6e11-46f5-936b-d1c87eae65ac)
![CheckedListBox - indeterminate state is correct in 5 0](https://github.com/dotnet/winforms/assets/113603457/e394ca15-fb00-4a06-a92b-52681b00b70e)

### Steps to reproduce

1. Create a Winforms .Net Core application with CheckedListBox.
2. Call `SetItemCheckState` method for one of CheckedListBox items.
3. Build and run application.
4. Launch Narrator to focus on that item.

Sample app:  [WinFormsApp1.zip](https://github.com/dotnet/winforms/files/11679033/WinFormsApp1.zip)

**Comment by merriemcgaw** [comment]
@dmitrii-drobotov do you know if UIA even supports the indeterminate toggle state? This is not a design pattern that anyone would suggest in today's world. It's either on or it's off.

**Comment by Eudora-Li01** [comment]
Verified this issue in the latest .NET 8.0 SDK build: 8.0.100-preview.7.23323.19, it was fixed. Indeterminate check box can be announced correctly with Narrator.
![image](https://github.com/dotnet/winforms/assets/130345015/2dc29162-4f17-4b69-93eb-da3c1032ddb9)

**Comment by MelonWang1** [comment]
Verified this on .NET 8.0 TP build: 8.0.100-preview.6.23328.9, issue was fixed, result is same as above test result.
